homeassistant/components/onvif/device.py

```diff
@@ -215,7 +215,7 @@ async def async_get_device_info(self) -> DeviceInfo:
         device_info = await device_mgmt.GetDeviceInformation()
 
         # Grab the last MAC address for backwards compatibility
-        mac = None
+        mac = device_info.HardwareId
```

Comment:
HardwareId is not guaranteed to be the MAC on all camera models. That’s why we make the network interfaces call in the first place. I’m ok using HardwareId as a fallback for Device ID, but not for MAC.